packages/sanity/src/core/studio/screens/RequestAccessScreen.tsx

```diff
@@ -0,0 +1,209 @@
+/* eslint-disable i18next/no-literal-string,@sanity/i18n/no-attribute-string-literals */
+import {Box, Button, Card, Dialog, Flex, Stack, Text, TextInput, useToast} from '@sanity/ui'
+import {useCallback, useEffect, useState} from 'react'
+import {
+  type CurrentUser,
+  getProviderTitle,
+  LoadingBlock,
+  type SanityClient,
+  useActiveWorkspace,
+} from 'sanity'
+
+import {NotAuthenticatedScreen} from './NotAuthenticatedScreen'
+
+interface AccessRequest {
+  id: string
+  status: 'pending' | 'accepted' | 'declined'
+  resourceId: string
+  resourceType: 'project'
+  createdAt: string
+  updatedAt: string
+  updatedByUserId: string
+  requestedByUserId: string
+  note: string
+}
+
+export function RequestAccessScreen() {
+  const [currentUser, setCurrentUser] = useState<CurrentUser | null>(null)
+  const [client, setClient] = useState<SanityClient | undefined>()
+  const [projectId, setProjectId] = useState<string | undefined>()
+  const toast = useToast()
+
+  const [error, setError] = useState<unknown>(null)
+  const [loading, setLoading] = useState(true)
+  const [isSubmitting, setIsSubmitting] = useState(false)
+
+  const [hasPendingRequest, setHasPendingRequest] = useState<boolean | undefined>()
+  const [hasTooManyRequests, setHasTooManyRequests] = useState<boolean | undefined>()
+
+  const [note, setNote] = useState<string>('')
+
+  const {activeWorkspace} = useActiveWorkspace()
+
+  const handleLogout = useCallback(() => {
+    activeWorkspace.auth.logout?.()
+  }, [activeWorkspace])
+
+  // Get config info from active workspace
+  useEffect(() => {
+    const subscription = activeWorkspace.auth.state.subscribe({
+      next: ({client: sanityClient, currentUser: user}) => {
+        // Need to get the client, projectId, and user from workspace
+        // because this screen is outside the SourceContext
+        setProjectId(sanityClient.config().projectId)
+        setClient(sanityClient.withConfig({apiVersion: '2024-07-01'}))
+        setCurrentUser(user)
+      },
+      error: setError,
+    })
+
+    return () => {
+      subscription.unsubscribe()
+    }
+  }, [activeWorkspace])
+
+  // Check if user has a pending
+  // access request for this project
+  useEffect(() => {
+    if (!client || !projectId) return
+    client
+      .request<AccessRequest[] | null>({
+        url: `/access/requests/me`,
+      })
+      .then((requests) => {
```

Comment:
Q: Might it be worth checking for any declined requests? I would imagine we could just fallback to `NotAuthenticatedScreen`

Reply:
Yup, just added that in now!